File: 0206-reverse-linked-list/0206-reverse-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
    struct ListNode *temp;
    temp=head;
    while(temp!=NULL){
        printf("%d",temp->val);
        temp=temp->next;
    }
    struct ListNode *prev;
    struct ListNode *nxt;
    struct ListNode *current=head;
    while(current!=NULL){
        if(current==head){
            nxt=current->next;
            prev=current;
            current->next=NULL;
            current=nxt;
        }
        if(current!=NULL){
            nxt=current->next;
            current->next=prev;
            prev=current;
            current=nxt;
        }
    }
    head=prev;
    
    return head;
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        temp=head
        prev=None
        nexti=None
        if temp is None:
            return head
        while temp is not None:
            nexti=temp.next
            temp.next=prev
            prev=temp
            temp=nexti
        return prev

        
       

        
        